src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -94,9 +106,17 @@ public void CompileMethod(MethodCodeNode methodCodeNodeNeedingCode)
                     }
                 }
 
+                IntPtr exception;
                 IntPtr nativeEntry;
                 uint codeSize;
-                _compile(_jit, _comp, ref methodInfo, flags, out nativeEntry, out codeSize);
+                JitWrapper(out exception, _jit, _comp, ref methodInfo, flags, out nativeEntry, out codeSize);
+                if (exception != IntPtr.Zero)
+                {
+                    char* szMessage = GetExceptionMessage(exception);
+                    string message = szMessage != null ? new string(szMessage) : "JIT Exception";
+                    FreeException(exception);
```

Comment:
Added a linked list to keep track of all exceptions, and free them in FlushPins (renamed to CompileMethodCleanup).